rasa/shared/core/training_data/story_writer/yaml_story_writer.py

```diff
@@ -45,32 +45,38 @@
 class YAMLStoryWriter:
     """Writes Core training data into a file in a YAML format. """
 
-    def dumps(self, story_steps: List[StoryStep]) -> Text:
+    def dumps(self, story_steps: List[StoryStep], append: bool = False) -> Text:
         """Turns Story steps into a string.
 
         Args:
             story_steps: Original story steps to be converted to the YAML.
+            append: Specify if step should be appended on stories list.
         Returns:
             String with story steps in the YAML format.
         """
         stream = yaml.StringIO()
-        self.dump(stream, story_steps)
+        self.dump(stream, story_steps, append)
         return stream.getvalue()
 
     def dump(
-        self, target: Union[Text, Path, yaml.StringIO], story_steps: List[StoryStep]
+        self,
+        target: Union[Text, Path, yaml.StringIO],
+        story_steps: List[StoryStep],
+        append: bool = False,
     ) -> None:
         """Writes Story steps into a target file/stream.
 
         Args:
             target: name of the target file/stream to write the YAML to.
             story_steps: Original story steps to be converted to the YAML.
         """
-        result = self.stories_to_yaml(story_steps)
+        result = self.stories_to_yaml(story_steps, append)
 
         rasa.shared.utils.io.write_yaml(result, target, True)
 
-    def stories_to_yaml(self, story_steps: List[StoryStep]) -> Dict[Text, Any]:
+    def stories_to_yaml(
```

Comment:
I think we should find an alternative way to append stories - `stories_to_yaml` returns a `Dict`, and shouldn't return list of story steps.